src/shallow-render-utils.ts

```diff
@@ -126,6 +126,52 @@ function isVNode(obj: any) {
   return Object(obj) === obj && typeof obj.type !== 'undefined';
 }
 
+/**
+ * Prevent double patching root VNodes which happens on rerender in some enzyme
+ * methods
+ */
+const patchCache = new WeakMap();
+
+/**
+ * Patch the root node of a render such that if it is a component:
+ * 1. We preserve its raw rendered output. So if it returns a Fragment, that
+ *    fragment shows up as an RSTNode
+ * 2. Calls to the component's render method are wrapped in the
+ *    `withShallowRendering` helper to ensure they continue to shallow render
+ */
+export function patchShallowRoot(root: VNode) {
```

Comment:
Thinking more about this, I want to try and make this a `withPatchedShallowRoot` method that un-does these changes after shallow rendering... Not sure how that'll work with enzyme's rerendering yet though. Need to dig more into it.